docs/docs/spec.md

```diff
@@ -290,7 +288,16 @@ All locations that are not explicitly listed as recognized are unrecognized.
 >     -   object creation expression
 >
 >         > For example, `new @Nullable ArrayList<String>()` has an unrecognized
->         > annotation.
+>         > annotation. However, note that type arguments in an object creation
+>         > expression can be annotated. For example, `new ArrayList<@Nullable
+>         > String>()` has a recognized annotation.
+>
+>     -   array creation expression
+>
+>         > For example, `new String @Nullable [5]` has an unrecognized
+>         > annotation. However, note that the component type in an array
+>         > creation expression can be annotated. For example, `new int [5]
+>         > @Nullable []` has a recognized annotation.
```

Comment:
I would be tempted to keep this example to a one-dimensional array because of how confusing the syntax gets. Do you think ghat `new @Nullable String[5]` could still get the point across while being a little simpler and closer to the earlier "bad" example?

Reply:
I agree.

Although it may be useful to add multidimensional array examples to the array component type entry, if only because the syntax is so odd. I wouldn't add it here.